libs/ml/include/ml/dataloaders/mnist_loaders/mnist_loader.hpp

```diff
@@ -93,14 +96,31 @@ class MNISTLoader : public DataLoader<LabelType, T>
     }
   }
 
-  void Display(T const &data) const
+  /**
+   * directly sets the data and labels variables
+   * This function must be implemented to override from Dataloader, but we
+   * may prefer to use SetupWithDataFiles helper function
+   * @param data
+   * @param label
+   * @return
+   */
+  bool AddData(InputType const &data, LabelType const &label) override
+  {
+    FETCH_UNUSED(data);
+    FETCH_UNUSED(label);
+    throw std::runtime_error(
+        "AddData not implemented for MNist example - please use Constructor or SetupWithDataFiles "
+        "methods");
+  }
+
+  void Display(InputType const &data) const
   {
     for (SizeType i{0}; i < FIGURE_WIDTH; ++i)
     {
       for (SizeType j{0}; j < FIGURE_HEIGHT; ++j)
       {
 
-        std::cout << (data.At(j, i, 0) > typename T::Type(0.5) ? char(219) : ' ');
+        std::cout << (data.At(j, i, 0) > typename InputType::Type(0.5) ? char(219) : ' ');
```

Comment:
I assume that char(219) is meant to print a black square? Apparently this is platform-dependent as the ASCII standard only goes up to 128. I'd suggest changing it to "#" (char(35)) or at least adding a comment!